astroquery/esa/jwst/tests/test_jwsttap.py

```diff
@@ -0,0 +1,1088 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+===============
+eJWST TAP tests
+===============
+
+European Space Astronomy Centre (ESAC)
+European Space Agency (ESA)
+
+"""
+import os
+import shutil
+from unittest.mock import MagicMock
+
+import astropy.units as u
+import numpy as np
+import pytest
+from astropy import units
+from astropy.coordinates.sky_coordinate import SkyCoord
+from astropy.table import Table
+from astropy.units import Quantity
+
+from astroquery.esa.jwst import JwstClass
+from astroquery.esa.jwst.tests.DummyTapHandler import DummyTapHandler
+from astroquery.ipac.ned import Ned
+from astroquery.simbad import Simbad
+from astroquery.utils import TableList
+from astroquery.utils.tap.conn.tests.DummyConnHandler import DummyConnHandler
+from astroquery.utils.tap.conn.tests.DummyResponse import DummyResponse
+from astroquery.utils.tap.core import TapPlus
+from astroquery.utils.tap.xmlparser import utils
+from astroquery.vizier import Vizier
+
+from astroquery.esa.jwst import conf
+
+
+def data_path(filename):
+    data_dir = os.path.join(os.path.dirname(__file__), 'data')
+    return os.path.join(data_dir, filename)
+
+
+def get_plane_id_mock(url, *args, **kwargs):
+    return ['00000000-0000-0000-879d-ae91fa2f43e2'], 3
+
+
+@pytest.fixture(autouse=True)
+def plane_id_request(request):
+    try:
+        mp = request.getfixturevalue("monkeypatch")
+    except AttributeError:  # pytest < 3
+        mp = request.getfuncargvalue("monkeypatch")
+    mp.setattr(JwstClass, '_get_plane_id', get_plane_id_mock)
+    return mp
+
+
+def get_associated_planes_mock(url, *args, **kwargs):
+    if kwargs.get("max_cal_level") == 2:
+        return "('00000000-0000-0000-879d-ae91fa2f43e2')"
+    else:
+        return planeids
+
+
+@pytest.fixture(autouse=True)
+def associated_planes_request(request):
+    try:
+        mp = request.getfixturevalue("monkeypatch")
+    except AttributeError:  # pytest < 3
+        mp = request.getfuncargvalue("monkeypatch")
+    mp.setattr(JwstClass, '_get_associated_planes', get_associated_planes_mock)
+    return mp
+
+
+def get_product_mock(params, *args, **kwargs):
+    if('file_name' in kwargs and kwargs.get('file_name') == 'file_name_id'):
+        return "00000000-0000-0000-8740-65e2827c9895"
+    else:
+        return "jw00617023001_02102_00001_nrcb4_uncal.fits"
+
+
+@pytest.fixture(autouse=True)
+def get_product_request(request):
+    try:
+        mp = request.getfixturevalue("monkeypatch")
+    except AttributeError:  # pytest < 3
```

Comment:
we are not supporting this old pytests, so you can drop the try/except

Reply:
Ok, done